src/compiler/types.ts

```diff
@@ -3679,6 +3679,7 @@ namespace ts {
         /*@internal*/init?: boolean;
         inlineSourceMap?: boolean;
         inlineSources?: boolean;
+        /*@internal*/isInferredProject?: boolean;

```

Comment:
As we discussed offline, if this is the right place for this (TBD), it should probably be called something like "allowJSOverwrite".